oecologia-australis.csl

```diff
@@ -268,42 +268,54 @@
           <choose>
             <if match="any" variable="number-of-pages publisher">
               <label variable="number-of-pages" form="short" plural="never"/>
-              <text variable="number-of-pages"/>
+              <text variable="number-of-pages" suffix="."/>
               <text variable="publisher" prefix="ed. " suffix=", "/>
               <text variable="publisher-place"/>
             </if>
           </choose>
         </group>
       </if>
       <else-if type="thesis" match="any">
+        <group suffix=".">
+          <text variable="genre" prefix=" "/>
+        </group>
         <group delimiter=", ">
           <text variable="publisher-place"/>
-          <text variable="publisher"/>
+          <text variable="publisher" prefix=" "/>
         </group>
       </else-if>
       <else-if type="post-weblog webpage" match="none">
-        <group delimiter=", ">
+        <group delimiter=": ">
           <choose>
             <if variable="event version" type="speech" match="none">
               <text variable="genre"/>
             </if>
           </choose>
           <choose>
-            <if type="article-journal article-magazine report" match="none">
-              <group delimiter=", ">
-                <text variable="publisher" font-variant="normal" prefix=" "/>
+            <if type="article-journal article-magazine report legislation" match="none">
+              <group delimiter=": ">
                 <choose>
-                  <if variable="publisher-place">
+                  <if type="legislation"/>
+                  <else-if match="any" variable="publisher-place">
                     <text variable="publisher-place" font-style="normal" prefix=" "/>
-                  </if>
+                  </else-if>
                   <else>
                     <text variable="event-place"/>
                   </else>
                 </choose>
+                <text variable="publisher" font-variant="normal" prefix=" "/>
               </group>
             </if>
           </choose>
           <text variable="number-of-pages" prefix="p. "/>
```

Comment:
The citation style is very weird here: Dunnett, N., & Kingsbury, N. 2008. Planting green roofs and living walls. 2nd ed. Portland, OR: Timber Press: p. 328.

reads like you're referring to page 328 of the book, while the variable `number-of-pages` refers to the total number of pages in the book. Are you sure you want that?

Reply:
I agree with you, but how I inform you before I am only the programmer. I don´t have an opinion on journal rules to do any modification. I will contact the editor and decide better these changes.

Reply:
Editorial board maintain "p. 328."